packages/grid/x-data-grid/src/colDef/gridSingleSelectColDef.tsx

```diff
@@ -1,15 +1,42 @@
 import { GRID_STRING_COL_DEF } from './gridStringColDef';
-import { GridColTypeDef } from '../models/colDef/gridColDef';
+import { GridColTypeDef, GridColDef } from '../models/colDef/gridColDef';
 import { renderEditSingleSelectCell } from '../components/cell/GridEditSingleSelectCell';
-import {
-  getGridSingleSelectOperators,
-  getGridSingleSelectQuickFilterFn,
-} from './gridSingleSelectOperators';
+import { getGridSingleSelectOperators } from './gridSingleSelectOperators';
+import { getLabelFromValueOption } from '../components/panel/filterPanel/filterPanelUtils';
+
+const isArrayOfObjects = (options: any): options is Array<{ value: any; label: string }> => {
```

Comment:
I think that the constraint of `{ value: any; label: string }` for option values is very limiting.
Currently, it's not possible to pass object structures different than the one mentioned. Any other structure should be transformed to fit `{ value: any; label: string }` shape *before* passing `valueOptions`  to the column.
I think we can improve the DX by allowing it to pass any object shape and extend the concept of `getOptionLabel` further:

```tsx
{
  type: 'singleSelect',
  // If the value option is an object and doesn't match the `{ value, label }` shape - throw an
  // error or log error message suggesting adding `parseValueOption`
  valueOptions: [
    { code: 'br', name: 'Brazil' },
    { code: 'it', name: 'Italy' }
  ],
  parseValueOption: ({ code, name }) => ({ value: code, label: name }),
}
```

Then we can operate on the `{ value, label }` shape internally while allowing custom any object shapes.
It's pretty similar to `getRowId`.

I know it would increase the scope of the PR, but we can take the opportunity to make breaking changes.
What do you think?

Reply:
> I would propose to do this in a follow-up if we want to do the right thing

Agree! But should we put this PR on hold then? Otherwise, we will first add `getOptionLabel` and then remove it in a few weeks. What do you think?

Reply:
We can prefix `getOptionLabel` with `unstable_` to make it easier to remove in the follow-up PR

Reply:
@m4theushw Should we proceed with `getOptionLabel`?

Reply:
So, instead of adding `GridColDef['parseOptionValue']` I proposed above, in the follow-up PRs we will add:
- `GridColDef['getOptionLabel']`
- `GridColDef['getOptionValue']`

Is this correct?

If so, then it sounds good to me 👍